design.md

```diff
@@ -2,12 +2,28 @@
 
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+The target audience for this design is students who are taking CS 5.  Because Picobot is the first assignment given to students, we aren’t assuming prior knowledge of maze-solving algorithms or of CS fundamentals in general.
+
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
 
+The main goals of our DSL design are as follows:
+- Introduce new CS students to programming basics such as ‘if’ statements, functions that take parameters, and to be explicit with their code.
+- Allow for easier organization of rules.  It makes sense to think of all the rules with the same start state as a single set of actions, so our design reflects this.
+- Incorporate natural language to make the learning curve a bit lower.
```

Comment:
While I agree this form lowers the learning curve, I think the bigger effect is this is just more a _intuitive_ way to do the syntax than the original way. Thus, a big benefit of your design is that you made it more _readable_!